src/conn/conn_api.c

```diff
@@ -1853,16 +1876,21 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
     }
 
     /*
-     * We own the database home, figure out if we're creating it. There are a few files created when
-     * initializing the database home and we could crash in-between any of them, so there's no
-     * simple test. The last thing we do during initialization is rename a turtle file into place,
-     * and there's never a database home after that point without a turtle file. If the turtle file
-     * doesn't exist, it's a create.
+     * If WiredTiger file exists but is size zero when it is not supposed to be (the turtle file
+     * exists and we are not salvaging), write a message but don't fail.
      */
-    WT_ERR(__wt_turtle_exists(session, &exist));
-    conn->is_new = exist ? 0 : 1;
+    empty = false;
+    if (fh) {
+        WT_ERR(__wt_filesize(session, fh, &size));
+        empty = (size_t)size == 0;
+        if (!is_salvage && !conn->is_new && empty)
+            WT_IGNORE_RET(__wt_msg(session, "WiredTiger version file is empty"));
+    }
 
-    if (conn->is_new) {
+    /*
+     * Populate WiredTiger file if new connection or WiredTiger file is empty and we are salvaging.
+     */
```

Comment:
Format as one line comment.

Reply:
Same here.

Reply:
Then I suggest expanding the comment to be a full, complete sentence that ends up multi-line. Something like:
Populate the WiredTiger file if this is a new connection or if the WiredTiger file is empty and we are salvaging.